### PR TITLE
Fixes #1515 - fixed encoding param

### LINKED ISSUE #1515: Typing "label:type-media" doesn't work in the homepage search form
Noticed by @magsout 

Steps to reproduce.
1. Go the homepage
2. click/tap on the search icon
3. Enter `label:type-media`

Expected:
Get the list of bugs with `type-media` label

Actual:
Empty list with a URL being 
`https://webcompat.com/issues?page=1&per_page=50&state=open&stage=all&sort=created&direction=desc&q=label%253Atype-media`

Note: `&q=label%253Atype-media`

Going to the issues list page and typing `label:type-media` in the search form is working and gives a URL
`https://webcompat.com/issues?page=1&per_page=50&state=open&stage=all&sort=created&direction=desc&q=label%3Atype-media`

but if I take this URI and I cut and paste it in the browser, I get back the gibberish URI from the home page
`https://webcompat.com/issues?page=1&per_page=50&state=open&stage=all&sort=created&direction=desc&q=label%253Atype-media` 
where the browser is trying to convert the percent into `%25`. So the progression goes indeed like this.

1. `label%3Atype-media`
2. `label%253Atype-media`
3. `label%25253Atype-media`

etc.

### PR DESCRIPTION
**Pull request by magsout**
r? @miketaylr 

I have fixed the issue #1515  and more.

- url: param q is encoded with `encodeURIComponent` to avoid `%25`
- input search is set when we refresh page if there is a `q` param
- `_currentSearch` is init directly in `this.issueList = new issueList.IssueView();`  to avoid double search at first page init.
- param `q` is removed of XHR rquest when search input is empty

~~There is still a bug. When you delete the search field. It is necessary to send a request (ok) and delete the param `q` (NOK, the test is here https://github.com/webcompat/webcompat.com/blob/master/webcompat/static/js/lib/issue-list.js#L464)~~

### Edit

Seems to fixe this issue #1023 

~~Last bug: Delete the request parameter https://github.com/webcompat/webcompat.com/blob/master/webcompat/static/js/lib/issue-list.js#L464~~

### CODE REVIEW

**Comment by miketaylr** [comment]
Whoa, awesome! Will have time to review this afternoon.

**Comment by miketaylr** [comment]
@magsout can you push a prettier error fix commit so tests can run?

**Comment by magsout** [comment]
>@magsout can you push a prettier error fix commit so tests can run?


Rah, need to add this issue https://github.com/webcompat/webcompat.com/issues/1474

**Comment by magsout** [comment]
Seems to fixe this issue https://github.com/webcompat/webcompat.com/issues/1023

**Comment by magsout** [comment]
@miketaylr hum, hum where is this commit https://github.com/webcompat/webcompat.com/commit/5998f12efa47be0dbb49b97ee85f5851414a1c82 because it fixed the bug like I did.. But I cannot find it..

**Comment by magsout** [comment]
ok, I get it. I find how to handle my last case, let me ush something

**Comment by miketaylr** [comment]
> hum, hum where is this commit 5998f12 because it fixed the bug like I did.. But I cannot find it..

Hmmm...... 👻

**Comment by magsout** [comment]
@miketaylr I add this https://github.com/webcompat/webcompat.com/pull/1518/files#diff-4ba6be3180115f78e14125480dcf4d7dR495

I don't know if it's clean or not.

**Comment by magsout** [comment]
Maybe it's better to make another PR, I fixed lot of things, what do you think @miketaylr  ?

**Comment by miketaylr** [comment]
> Maybe it's better to make another PR, I fixed lot of things, what do you think @miketaylr ?

There's not that much code changing in these 4 commits, I think it should be fine here.

File: webcompat/static/js/lib/issue-list.js [hunk 1]
```diff
@@ -336,14 +336,18 @@ issueList.IssueView = Backbone.View.extend(
       // There are some params in the URL
       if (urlParams.length !== 0) {
         queryMatch = urlParams.match(this._searchRegex);
+        if (queryMatch) {
```
Comment (miketaylr) [inline webcompat/static/js/lib/issue-list.js]: I wonder if there's a difference in behavior for logged-in or not logged-in users...

**Comment by miketaylr** [comment]
@magsout this looks good, but I'd like to write some tests as well. Can you help me think of the scenarios we should test?

1. have a `type-media` label in the test repo, make sure that `label:type-media` search query returns a result
2. have a q param with `foo`, and the search input should load with `foo`

> param q is removed of XHR rquest when search input is empty

How can we test this one?

**Comment by magsout** [comment]
@miketaylr is it enought if:

3) have a q param with something With 0 result
4) remove q param and have result again ?

**Comment by magsout** [comment]
>r+, but let's write some tests before merging. I'm happy to write the tests, unless you want to

@miketaylr  unless it can be wait one or two weeks, be my guest ;)

File: webcompat/static/js/lib/issue-list.js (hunk 1 above)
Comment (magsout) [inline webcompat/static/js/lib/issue-list.js]: Hum, I don't think so..

**Comment by miketaylr** [comment]
:))))) I'll work on these over the next few days.

**Comment by miketaylr** [comment]
> @miketaylr is it enought if:

Another way would just be to load a page with `q=thing`, then search for `""` and check that `q=thing` isn't in the URL.

**Comment by miketaylr** [comment]
@magsout just pushed some tests, will likely merge tomorrow morning (don't want to break the universe at the end of the day, in case something goes wrong). ❤️

**Pull request merged**